lib/iris/quickplot.py

```diff
@@ -61,9 +61,12 @@ def _title(cube_or_coord, with_units):
     return title
 
 
-def _label(cube, mode, result=None, ndims=2, coords=None):
+def _label(cube, mode, result=None, ndims=2, coords=None, axes=None):
     """Puts labels on the current plot using the given cube."""
 
+    if axes is not None:
+        plt.sca(axes)
```

Comment:
Just so we are all clear, using matplotlib's state machine is really unpleasant. However, since iris is already doing it left-right-and-center in this code, I'm OK with this. I might be even more comfortable if we were restoring the current axes to what it was after the function finishes...

Reply:
I was a bit uncertain about the sensibility of using `plt.sca`. I will add a couple lines to set the axes back to what they were before.